tiledb/sm/subarray/subarray.cc

```diff
@@ -1030,8 +1022,8 @@ Status Subarray::compute_est_result_size(
     max_size_fixed =
         utils::math::safe_mul(cell_num, array_schema->cell_size(attr_name));
   }
-  ret.size_fixed_ = MIN(ret.size_fixed_, max_size_fixed);
-  ret.size_var_ = MIN(ret.size_var_, max_size_var);
+  ret.size_fixed_ = std::min(ret.size_fixed_, (double)max_size_fixed);
+  ret.size_var_ = std::min(ret.size_var_, (double)max_size_var);
```

Comment:
Don't use a `(double)` cast here, use `std::min<double>(...)` instead.

Reply:
Which approach should be followed when types of compared values are mismatched? I mean C-style casting will be replaced by your suggestion but what should be template-type in this case?

Reply:
The template type should be the type of the lvalue that `std::min` is assigned. In this case, both `ret.size_fixed_` and `ret. size_var_` are of type `double`. I'm not implying that it is safe to interpret the uint64_t argument as a double, but it maintains the existing behavior.

Reply:
Could you please review these changes again? I fixed them way that you suggested.